### PR TITLE
Deprecate series.nonzero (GH18262)

### PR DESCRIPTION
**Pull request by makbigc**
xref #18262

-  passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
-  whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
this would need a test. Also you would need to remove any uses of this from the test code itself (otherwise it would show a warning)

File: pandas/core/series.py [hunk 1]
```diff
@@ -564,6 +566,9 @@ def nonzero(self):
         --------
         numpy.nonzero
         """
+        msg = ("Series.nonzero() is deprecated "
```
Comment (jreback) [inline pandas/core/series.py]: can you show what it is replaced by here

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
can you merge master and update

File: pandas/core/frame.py [hunk 2]
```diff
@@ -4480,7 +4480,10 @@ def dropna(self, axis=0, how='any', thresh=None, subset=None,
                 else:
                     raise TypeError('must specify how or thresh')
 
-            result = self._take(mask.nonzero()[0], axis=axis)
+            if axis == 0:
```
Comment (jreback) [inline pandas/core/frame.py]: you can do
``result = self.loc(axis=axis)[mask]``

File: pandas/core/frame.py [hunk 3]
```diff
@@ -4515,7 +4518,7 @@ def drop_duplicates(self, subset=None, keep='first', inplace=False):
         duplicated = self.duplicated(subset, keep=keep)
 
         if inplace:
-            inds, = (-duplicated).nonzero()
+            inds, = (-duplicated).values.nonzero()
```
Comment (jreback) [inline pandas/core/frame.py]: change this like
``(~duplicated)._values.nonzero()``

**Review by jreback — CHANGES_REQUESTED** [review]
are there any warnings issued by the test suite? need to change any internal uses of this

**Comment by jreback** [comment]
also remove this from api.rst if it exists

**Review by jreback — CHANGES_REQUESTED** [review]
there are NO warnings from the test suite?

File: pandas/io/stata.py [hunk 4]
```diff
@@ -1629,7 +1629,8 @@ def _do_convert_missing(self, data, convert_missing):
                 continue
 
             if convert_missing:  # Replacement follows Stata notation
-                missing_loc = np.argwhere(missing)
+
+                missing_loc = np.argwhere(missing.values)
```
[the remark this answers is not in the document]
Comment (makbigc) [reply pandas/io/stata.py]: when nonzero is implemented in Series, np.argwhere will call it

**Comment by makbigc** [comment]
No warning is raised due to deprecation of Series.nonzero, running the test suite in my own machine.

File: pandas/core/series.py [hunk 5]
```diff
@@ -542,6 +542,8 @@ def nonzero(self):
         """
         Return the *integer* indices of the elements that are non-zero.
 
+        .. deprecated:: 0.24.0
```
Comment (jreback) [inline pandas/core/series.py]: can you add what this is replaced here as well

**Review by jreback — CHANGES_REQUESTED** [review]
looks fine, but need to remove doc/source/api.rst which you seem to have added back

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thank @makbigc

**Pull request merged**